`pad` gives wrong results version>=0.4.36

### Description

This snippet started failing after 0.4.36 and still fails in 0.5.1:

```python
import jax
import numpy as np

x = np.arange(1, 10).reshape(3,3)
np.testing.assert_allclose(
    np.pad(x, mode="constant", pad_width=3, constant_values=(1, 2)),
    jax.numpy.pad(x, mode="constant", pad_width=3, constant_values=(1, 2)),
)
```
```
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0
Mismatched elements: 9 / 81 (11.1%)
Max absolute difference among violations: 1
Max relative difference among violations: 0.5
 ACTUAL: array([[1, 1, 1, 1, 1, 1, 2, 2, 2],
       [1, 1, 1, 1, 1, 1, 2, 2, 2],
       [1, 1, 1, 1, 1, 1, 2, 2, 2],...
 DESIRED: array([[1, 1, 1, 1, 1, 1, 2, 2, 2],
       [1, 1, 1, 1, 1, 1, 2, 2, 2],
       [1, 1, 1, 1, 1, 1, 2, 2, 2],...
```

### System info (python version, jaxlib version, accelerator, etc.)

```
platform: uname_result(system='Linux', node='fedora', release='6.12.11-200.fc41.x86_64', version='#1 SMP PREEMPT_DYNAMIC Fri Jan 24 04:59:58 UTC 2025', machine='x86_64')
```

--- a/jax/_src/numpy/lax_numpy.py
+++ b/jax/_src/numpy/lax_numpy.py
@@ -3986,12 +3986,7 @@ def _pad_constant(array: Array, pad_width: PadValue[int], constant_values: Array
     if constant_values.shape[-1] == 1:
       widths = [(low, high, 0) for (low, high) in pad_width]
       return lax.pad(array, squeeze(constant_values), widths)
-    elif constant_values.shape[-1] == 2:
-      widths = [(low, 0, 0) for (low, _) in pad_width]
-      array = lax.pad(array, constant_values[0], widths)
-      widths = [(0, high, 0) for (_, high) in pad_width]
-      return lax.pad(array, constant_values[1], widths)
-    else:
+    elif constant_values.shape[-1] != 2:
       raise ValueError("jnp.pad: constant_values has unsupported shape "
                       f"{constant_values.shape}. If the shape is 1D or 2D, the "
                       "last dimension must be of size 1 or 2.")

--- a/tests/lax_numpy_test.py
+++ b/tests/lax_numpy_test.py
@@ -1143,7 +1143,8 @@ class LaxBackedNumpyTests(jtu.JaxTestCase):
           # (constant,)
           (0,), (2.718,),
           # ((before_const, after_const),)
-          ((0, 2),), ((-1, 3.14),),
+          (0, 2),
+          (-1, 3.14),
           # ((before_1, after_1), ..., (before_N, after_N))
           tuple((i / 2, -3.14 * i) for i in range(len(shape))),
       ]

